### PR TITLE
[16.0][MIG] fieldservice_sale

### PR DESCRIPTION
**Pull request by epanisset**

### CODE REVIEW

File: fieldservice_sale/models/sale_order.py [hunk 1]
```diff
@@ -41,25 +39,25 @@ def onchange_partner_id(self):
...
-        self.fsm_location_id = location_ids and location_ids[0] or False
+        location = self.partner_id.fsm_location
+        if not location:
+            domain = [
+                "|",
+                "|",
+                ("partner_id", "=", self.partner_id.id),
+                ("partner_id", "=", self.partner_shipping_id.id),
+                ("partner_id", "=", self.partner_id.commercial_partner_id.id),
+            ]
```
Comment (yankinmax) [inline fieldservice_sale/models/sale_order.py]: can you pls use OR expression from osv?
```
expression.OR(
    [
        [("partner_id", "=", self.partner_id.id)],
        [("partner_id", "=", self.partner_shipping_id.id)],
        [("partner_id", "=", self.partner_id.commercial_partner_id.id)],
    ]
)
```

**Comment by yankinmax** [comment]
@SilvioC2C this can cause Travis failure not related to this module
`ERROR odoo odoo.schema: Table 'fsm_location': unable to set NOT NULL on column 'inventory_location_id'`
I suggest to open one more PR to set required attribute directly in view for this field.
If not - LG

**Comment by SilvioC2C** [comment]
@yankinmax that shouldn't cause Travis to fail; moreover, that field is defined in `fieldservice_stock`, not `fieldservice_sale`.

**Comment by yankinmax** [comment]
@SilvioC2C I completely agree with you, but modules are installed altogether in the repo, so I had such situations.

**Comment by SilvioC2C** [comment]
@yankinmax that's ok, but you should open an issue for that or mention this in the proper PR, this is the wrong place :smile:

**Comment by yankinmax** [comment]
@SilvioC2C I've opened a request on the migration PR for `fieldservice_stock`:
https://github.com/OCA/field-service/pull/993#issuecomment-1410106116

File: fieldservice_sale/models/sale_order.py (hunk 1 above)
Comment (SilvioC2C) [reply fieldservice_sale/models/sale_order.py]: Why? We're not composing domain dinamically, so we know beforehand which conditions are to be chained with an `OR`, so we can create the domain directly using the `"|"` operator. It's easier to read, and doesn't force Odoo to compute a new domain.

**Review by SilvioC2C — APPROVED** [review]

**Comment by brian10048** [comment]
Would you be able to add #1050 in this migration?

**Review by Camille0907 — APPROVED** [review]
LGTM

**Comment by dalonsod** [comment]
@epanisset would you resume this PR? In that case, please add #1072 if needed. Thanks!

**Review by stefan-tecnativa — APPROVED** [review]
Functionally tested 👍️. @epanisset what is the state of this PR?

**Comment by pedrobaeza** [comment]
Superseded by #1101, but respecting commit attribution.

**Pull request closed without merging**